added httpMethod in perform action parameter and default to http.Get

xavave opened the pull request.

Commit:
update httppost  to http get

Commit:
added httpMethod in performaction parameter

Commit:
updated .gitignore

AmenJlili reviewed (approved):
Should be good... nothing to see here

AmenJlili commented on src/LicenseManager.cs:
```diff
@@ -222,30 +224,31 @@ public ILicenseResponse PerformAction(Consts.Action action, License license)
         {
-            return Task.Run(() => {
+            return Task.Run(() =>
+            {
```
@xavave  I think you need to inject httpMethod synchronous method

AmenJlili commented on Tests/WPLMNETClient.NetFramework.Tests/Tests.cs:
```diff
@@ -18,19 +18,19 @@ public void OnStartUp()
         {
 
```
We should prolly add a little  comment to the user the configuration and license object are specific to their website

Commit:
modified configuration test values +parsed date before sending to server in order to avoid null values in sent dates
  restRequest.AddParameter("date_created", license.DateCreated.ToString("yyyy-MM-dd"));
   restRequest.AddParameter("date_renewed", license.DateRenewed.ToString("yyyy-MM-dd"));
   restRequest.AddParameter("date_expiry", license.DateExpired.ToString("yyyy-MM-dd"));

xavave commented:
I'm not sure to fully understand your change request: anyway, I've modified configuration parameters (which were specific to your debugging server) 
I've also I've found and fixed another  issue:  the 3 dates sent to server (date_created,date_renewed,date_expiry) were sent empty, because server is waiting for this date type: yyyy-MM-dd so I've parsed these datetimes with ToString("yyyy-MM-dd") and, now the dates are properly sent to the server.

the first issue was not related to synchronous or asynchronous method (I was already using synchronous method), but due to the fact that my client App (a WPF app) is not located on the wordpress server (which is on my synology NAS) so I guess that when you want to POST method from an external location to the wordpress server, wordpress POST client request needs to be authentified , whereas with GET method , no need to authenticate with a wordpress account /token before sending the request

Commit:
updated Nuget packages: restsharp + newtonsoft.Json

Commit:
upadte nuget packages for test + migrate package.Config to package referencce

Commit:
fix productreference

Commit:
added Idisposable interface for configuration class + fix product_ref restsharp parameter

Commit:
fix: subscriber id restsharp parameter name

Commit:
// add registered domain to rest sharp params

Commit:
added license Id --> added update license with create

Commit:
added fields: mode, version, ipaddress

jliliamen commented:
I'm talking about this : 
![image](https://user-images.githubusercontent.com/16106587/113791059-a38dcd80-96f7-11eb-8885-e692d26878ef.png)
You forget to pass the optional parameter... 

I'll fix that myself no problem. Thanks for your commits! Merci beaucoup!
